Dmin_autocalculator.py

```diff
@@ -0,0 +1,230 @@
+    try:
+        n_points = config["DIFFUSION_TYPE_PARAM"]["number_of_points"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter number_of_points missing in config.")
+    try:
+        MSD_f_area = config["DIFFUSION_TYPE_PARAM"]["MSD_fit_area"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter MSD_fit_area missing in config.")
+    try:
+        dof = config["DIFFUSION_TYPE_PARAM"]["degree_of_freedom"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter degree_of_freedom missing in config.")
+    # minimum detectable D should be zero to calculated  this value
+    min_dD = 0
+    try:
+        min_tra_len = config["DIFFUSION_TYPE_PARAM"]["min_track_length"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter min_track_length missing in config.")
+
+    try:
+        id_type = config["ID_TYPE"]["id_type"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter id_type missing in config.")
+    try:
+        min_traj = config["FILTER_PARAMETERS"]["min_trajectory_len"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter min_trajectory_len missing in config.")
+    try:
+        max_traj = config["FILTER_PARAMETERS"]["max_trajectory_len"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter max_trajectory_len missing in config.")
+    try:
+        min_D = config["FILTER_PARAMETERS"]["min_D"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter min_D missing in config.")
+    try:
+        max_D = config["FILTER_PARAMETERS"]["max_D"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter max_D missing in config.")
+    try:
+        filter_immobile = config["FILTER_PARAMETERS"]["filter_for_immobile"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter filter_for_immobile missing in config.")
+    try:
+        filter_confined = config["FILTER_PARAMETERS"]["filter_for_confined"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter filter_for_confined missing in config.")
+    try:
+        filter_free = config["FILTER_PARAMETERS"]["filter_for_free"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter filter_for_free missing in config.")
+    try:
+        filter_noType = config["FILTER_PARAMETERS"]["filter_for_noType"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter filter_for_noType missing in config.")
+    try:
+        save_dir = config["SAVE_DIR"]["sav"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter sav missing in config.")
+    # resets tracking directories
+    try:
+        os.mkdir(save_dir)
+    except FileExistsError:
+        pass
+    try:
+        os.mkdir(save_dir + '\\trackAnalysis')
+        os.mkdir(save_dir + '\\trackAnalysis\\backgrounds')
+        os.mkdir(save_dir + '\\trackAnalysis\\cells')
+    except FileExistsError:
```

Comment:
print a statement that {directory} exists and will be overwritten → users should be informed when data is overwritten.